process/rewardTransaction/interceptedRewardTransaction_test.go

```diff
@@ -317,6 +318,88 @@ func TestNewInterceptedRewardTransaction_CheckValidityShouldWork(t *testing.T) {
 	assert.Nil(t, err)
 }
 
+func TestNewInterceptedRewardTransaction_IsForCurrentShard(t *testing.T) {
+	t.Parallel()
+
+	receiverAddress := []byte("receiver address")
+	testShardID := uint32(2)
+	value := big.NewInt(100)
+	rewTx := rewardTx.RewardTx{
+		Round:   0,
+		Epoch:   0,
+		Value:   value,
+		RcvAddr: receiverAddress,
+	}
+
+	mockShardCoordinator := &mock.ShardCoordinatorStub{}
+	marshalizer := &mock.MarshalizerMock{}
+	txBuff, _ := marshalizer.Marshal(&rewTx)
+	t.Run("same shard ID with the receiver should return true", func(t *testing.T) {
+		mockShardCoordinator.ComputeIdCalled = func(address []byte) uint32 {
+			if bytes.Equal(address, receiverAddress) {
+				return testShardID
+			}
+
+			return 0
+		}
+		mockShardCoordinator.SelfIdCalled = func() uint32 {
+			return testShardID
+		}
+
+		irt, err := rewardTransaction.NewInterceptedRewardTransaction(
+			txBuff,
+			marshalizer,
+			&hashingMocks.HasherMock{},
+			createMockPubkeyConverter(),
+			mockShardCoordinator)
+		assert.Nil(t, err)
+
+		assert.True(t, irt.IsForCurrentShard())
+	})
+	t.Run("metachain should return true", func(t *testing.T) {
+		mockShardCoordinator.ComputeIdCalled = func(address []byte) uint32 {
+			if bytes.Equal(address, receiverAddress) {
+				return testShardID
+			}
+
+			return 0
+		}
+		mockShardCoordinator.SelfIdCalled = func() uint32 {
+			return core.MetachainShardId
+		}
+
+		irt, err := rewardTransaction.NewInterceptedRewardTransaction(
+			txBuff,
+			marshalizer,
+			&hashingMocks.HasherMock{},
+			createMockPubkeyConverter(),
+			mockShardCoordinator)
+		assert.Nil(t, err)
+		assert.True(t, irt.IsForCurrentShard())
+	})
+	t.Run("different shard should return true", func(t *testing.T) {
```

Comment:
still true here? The check is with false: assert.False(t, irt.IsForCurrentShard())

Reply:
fixed